Source/Dafny/Dafny.atg

```diff
@@ -2832,7 +2848,9 @@ ForallStmt<out Statement/*!*/ s>
   .)
 
   {
-    "ensures" Expression<out e, false, true>  (. ens.Add(new AttributedExpression(e)); .)
+    "ensures"                       (. Attributes ensAttrs = null; .)
```

Comment:
See comment above. The various `...Attrs` variables in this production can also be compiled into one.
Or, if (here and above) they are kept as separate variables, then each one could be introduced like you do in L2851 and L3466 (as opposed to like in L2067), but still all be called `attrs`.

Reply:
Changed here and elsewhere